src/azul/drs.py

```diff
@@ -74,10 +87,133 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+class DRSURI(metaclass=ABCMeta):
+    scheme: str
+    netloc: str
+    object_id: str
+
+    @classmethod
+    def parse(cls, drs_uri: str) -> 'DRSURI':
+        try:
+            drs_uri = RegularDRSURI.parse(drs_uri)
+        except ValueError as e:
+            if len(e.args) == 1 and isinstance(e.args[0], str) and e.args[0].startswith('Invalid port '):
+                drs_uri = CompactDRSURI.parse(drs_uri)
+            else:
+                raise
+        return drs_uri
+
+    @abstractmethod
+    def expand(self, client: 'DRSClient') -> mutable_furl:
+        raise NotImplementedError
+
+    def _check_scheme(self) -> None:
+        assert self.scheme == 'drs', self
+
+
+class RegularDRSURI(DRSURI, furl):
```

Comment:
I considered this, too, but don't think it's a good idea. Inheriting from `furl` increases the surface area of this class tremendously. Only expose what is needed at the usage sites.